### PR TITLE
Fix: Fixed issue where some file operations would fail

### LINKED ISSUE #15133: Bug: Permanently delete doesn't work all the time
### Description

Currently when trying the prompt show and the permanently delete box is check after running the file/folder will disappear and Status center show it was successful. But refreshing the folder will show the file/folder again. 
Sometimes it also only delete the bottom most selected file and no the others.

### Steps To Reproduce

1. Select a file/folder
2. Shift delete
3. See prompt shown with check mark enabled
4. See file/folder looks deleted
5. Refresh folder

### Requirements

- permanently delete file/folder when checked

### Files Version

3.3.1.0

### Windows Version

10.0.22621.3374

### Log File

[debug.log](https://github.com/files-community/Files/files/14895620/debug.log)

### PR DESCRIPTION
**Pull request by hishitetsu**
**Resolved / Related Issues**
- [X] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #15133 

**Validation**
How did you test these changes?
- [X] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?

### CODE REVIEW

**Comment by gumbarros** [comment]
LGTM to me, the changed method is the same from my debug.log from previous issues:
```
2024-04-05 19:35:03.0538|Error|System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. ()
 ---> System.Runtime.InteropServices.COMException (0x800706F4)
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|39_0(Int32 hr)
   at ABI.Files.App.Server.Database.IFileTagsDatabaseClassMethods.SetTags(IObjectReference _obj, String filePath, Nullable`1 frn, String[] tags)
   at Files.App.Utils.FileTags.FileTagsHelper.UpdateTagsDb()
   ```

File: src/Files.App/Data/Items/ListedItem.cs [hunk 1]
```diff
@@ -102,7 +102,7 @@ public string[] FileTags
 				{
 					Debug.Assert(value != null);
 					var dbInstance = FileTagsHelper.GetDbInstance();
-					dbInstance.SetTags(ItemPath, FileFRN, value);
+					dbInstance.SetTags(ItemPath, FileFRN, value ?? []);
```
Comment (hez2010) [inline src/Files.App/Data/Items/ListedItem.cs:105]: We already asserted `value` is not null. I think we can keep here as it was.

**Review by hez2010 — APPROVED** [review]
Otherwise LGTM!

File: src/Files.App/Data/Items/ListedItem.cs (hunk 1 above)
Comment (hishitetsu) [reply src/Files.App/Data/Items/ListedItem.cs:105]: `Debug.Assert` is ignored in the release build, so I'm not sure if `value` is not null here.

**Pull request merged**